HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,65 @@
+from pyspark import SparkContext
+
+# Function to execute single-source BFS
+def ssBFS(graph, sourceNode, sc, diameter=10):
+	# Take graph and convert it to the following structure:
+	# (char, [neighbors, depth, visited])
+	depthGraph = graph.map(lambda (char, neighbs): (char, [neighbs, 0 if char == sourceNode else float("inf"), False]))
+	
+	# Used to be a fixed loop for diameter size, but accumulator makes 
+	# this unnecessary, so making this an inf loop
+	while True:
+	# for i in range(diameter):
+		# This is our queue size. If it ever stays 0 that means we're not
+		# adding to the queue anymore and can kick out
+		queue = sc.accumulator(0)
+
+		# Execute a single search step and store the new graph 
+		# (char, [[neighb1, neighb2, ...], depth, visited])
+		stepThrough = depthGraph.flatMap(lambda (char, neighbInfo): executeSearchStep(sourceNode, char, neighbInfo, queue))
+		# print stepThrough.take(3)
+		# print "------------"
+
+		# Combine touched neighbors of given character through a simple reduction. For every 2 neighbors
+		# combine their respective neighbors into a single de-duped set-list, find the minimum depth, 
+		# and mark visited as true if at least one was visited. This way we can continue traversing as before.
+		# (char, [[neighb1, neighb2, ...], depth, visited])
+		combinedNeighbs = stepThrough.reduceByKey(lambda n1, n2: ([list(set(n1[0] + n2[0])), min(n1[1], n2[1]), n1[2] or n2[2]]))
+		# print combinedNeighbs.take(3)
+		# print "############"
```

Comment:
Yeah, these were tests I ran prior to submission that I chose to omit but not delete right before I pushed. Is this bad practice? Should I have left these as uncommented code or removed them entirely?